src/store/modules/SplashModule.ts

```diff
@@ -0,0 +1,158 @@
+import RequestItem from 'src/model/requests/RequestItem';
+import { ActionTree } from 'vuex';
+import { State as RootState } from 'src/store';
+import type { PackageListIndex } from 'src/store/modules/TsModsModule';
+
+interface State {
+    requests: RequestItem[];
+    splashText: string;
+}
+
+/**
+ * State for Profiles, i.e. list for profiles in a single game/community.
+ */
+export const SplashModule = {
+    namespaced: true,
+
+    state: (): State => ({
+        requests: [],
+        splashText: '',
+    }),
+    mutations: {
+        resetRequestProgresses(state: State) {
+            state.requests.forEach((request) => request.setProgress(0));
+        },
+        initialiseRequests(state: State) {
+            state.requests = [
+                new RequestItem('UpdateCheck', 0),
+                new RequestItem('PackageListIndex', 0),
+                new RequestItem('PackageListChunks', 0),
+                new RequestItem('Vuex', 0)
+            ];
+        }
+    },
+    actions: <ActionTree<State, RootState>>{
+        setSplashText({ state }, splashText: string) {
+            state.splashText = splashText;
+        },
+        getRequestItem({state}, requestName: string): RequestItem {
+            const item = state.requests.find((ri) => ri.getName() === requestName);
+
+            if (item === undefined) {
+                throw new Error(`Unknown RequestItem "${requestName}"`);
+            }
+
+            return item;
+        },
+        async getThunderstoreMods({commit, dispatch}) {
+            commit('tsMods/startThunderstoreModListUpdate', null, {root: true});
+            const hasPriorCache = await dispatch('doesGameHaveLocalCache');
+
+            if (!hasPriorCache) {
+                const packageListIndex = await dispatch('fetchPackageListIndex');
+                const areAllChunksProcessedSuccessfully = await dispatch('fetchPackageListChunksIfUpdated', packageListIndex);
+                await dispatch('pruneRemovedMods', packageListIndex, areAllChunksProcessedSuccessfully);
+            }
+
+            await dispatch('triggerStoreModListUpdate');
+            commit('tsMods/finishThunderstoreModListUpdate', null, {root: true});
+
+            if (hasPriorCache) {
+                // We want this to trigger in the background to ensure that the user gets an up-to-date modlist as soon as possible.
+                dispatch('tsMods/syncPackageList', null, {root: true});
+            }
+
+            // TODO - Move to next screen (name: profiles)
```

Comment:
Misleading, this is done in the Vue component (which feels like the correct place for it) so drop the comment